workerapi/worker.thrift

```diff
@@ -0,0 +1,38 @@
+enum Status {
+  UNKNOWN = 0
+  READY = 1
+  PENDING = 2
+  FAILURE = 3
```

Comment:
What if you get status FAILURE but ExitCode 0? Or vice versa?

You can't tell aborted and timeout apart in that way, so that's why they should be separate states.